master/buildbot/status/status_stash.py

```diff
@@ -84,17 +93,42 @@ def __init__(self, base_url, user, password):
             base_url += '/'
         self.base_url = '%srest/build-status/1.0/commits/' % (base_url, )
         self.auth = b64encode('%s:%s' % (user, password))
+        self.key_format = key_format
+        self.name_format = name_format
         self._sha = Interpolate('%(src::revision)s')
+        self._branch = Interpolate('%(src::branch)s')
+        self._buildnumber = Interpolate('%(prop:buildnumber)s')
         self.master_status = None
 
     @defer.inlineCallbacks
+    def _send(self, request_kwargs, body, error_message):
```

Comment:
no need inlinecallbacks if you keep it written with addCallback and addErrback.

just return d